core/src/block_strider/subscriber/mod.rs

```diff
@@ -420,3 +421,59 @@ pub mod test {
         }
     }
 }
+
+pub async fn find_longest_diffs_tail(mc_block: BlockId, storage: &Storage) -> Result<usize> {
+    let mc_block_stuff = load_mc_block_stuff(mc_block, storage).await?;
+
+    let shard_block_handles = load_shard_block_handles(&mc_block_stuff, storage).await?;
+
+    let mut min_tail_len = None;
+
+    for block_handle in shard_block_handles {
+        let block = storage
+            .block_storage()
+            .load_block_data(&block_handle)
+            .await?;
+        let tail_len = block.block().out_msg_queue_updates.tail_len as usize;
+
+        min_tail_len = Some(min_tail_len.map_or(tail_len, |min: usize| min.min(tail_len)));
```

Comment:
Why do we get minimal tail length from shards? Should it be the longest tail instead?